ddtrace/_trace/span.py

```diff
@@ -60,9 +60,9 @@
 class SpanEvent:
     __slots__ = ["name", "attributes", "time_unix_nano"]
 
-    def __init__(self, name: str, attributes: Optional[Dict[str, str]] = None, time_unix_nano: Optional[int] = None):
+    def __init__(self, name: str, attributes: Optional[Dict[str, Any]] = None, time_unix_nano: Optional[int] = None):
         self.name: str = name
-        self.attributes: Dict[str, str] = attributes or {}
+        self.attributes: Dict[str, Any] = attributes or {}
```

Comment:
SpanEvent.Attributes must be a dict with string key value pairs. We should ensure the key is a string before it's added to the SpanEvent object. This will simplify the encoding logic.